### DISCUSSION TITLE
Any way to find out the ifc type?

### DISCUSSION THREAD

**Discussion by olazt** [IfcOpenShell/IfcOpenShell#2707] [Q&A]
Hello,
We are developing a routine for downloading lots of ifc files from one system and using them in GIS. Since it will be a batch process and files are coming from different sources I wonder if ifcopenshell has any function which allows us to read "type" property from an .ifc file?

I see that all examples in tutorial are based on type, which one defines by using e.g. "model.by_type('IfcWall')", but what if I do not know  to what type (IFC Entity) my objects are assigned to and want to find it out first, in order to use mentioned function in second step to get the other attributes of the objects?

Has anyone got any tips? I will appreaciate any help

**Comment by aothms** [comment]
`ifcopenshell.entity_instance` defines the is_a() method to return the type as a string.

`set(inst.is_a() for inst in my_file)` will for example give you all types. Keep in mind that there might be in rare cases inheritance relationships between those types (most concrete entities in IFC are leafs but not all). In that case it might be wise to do `by_type(include_subtypes=False)`
